4 4 multithreads

IlyaUman opened the pull request.

IlyaUman committed:
Two exercises about calculator

IlyaUman committed:
Second task about timer

в качестве аргументов ожидается строка вида 1h 2m 3s или 2m 3s или 3s или пустая строка. Таймер сработает в любом случае не более чем через 5 секунд, сделано специально, чтобы можно было проверить ввод минут и часов и не ждать.

IlyaUman committed:
Multithreads

Dokril commented on 4-4-multithreads/index.js:
```diff
@@ -0,0 +1,79 @@
+const arr2 = arr.slice(Math.floor(arr.length / 4), Math.floor(arr.length / 2 ))
+const arr3 = arr.slice(Math.floor(arr.length / 2), Math.floor(arr.length / 4 * 3))
+const arr4 = arr.slice(Math.floor(arr.length / 4 * 3), arr.length )
+
```
Надо массив arr разбить на массив массивов. В зависимости от числа ядер 
```ts
import { cpus } from 'node:os'
cpus().length
```
[1,2,3,4,5,6]
[[1,2,3],[4,5,6]]

Dokril commented on 4-4-multithreads/index.js:
```diff
@@ -0,0 +1,79 @@
+		if (el % 3 == 0) {
+			num++
+		}
+	})
```
Лушче вынести в отдельную функцию и использовать и в воркери и тут

IlyaUman committed:
Second task about timer

в качестве аргументов ожидается строка вида 1h 2m 3s или 2m 3s или 3s или пустая строка. Таймер сработает в любом случае не более чем через 5 секунд, сделано специально, чтобы можно было проверить ввод минут и часов и не ждать.

IlyaUman committed:
Multithreads
